### PR TITLE
[WIP] build: add cargo feature to drop various tools

### PR DESCRIPTION
**Pull request by bartlomieju**
This is really a prototype and is not intended to be landed.

This commit adds default cargo feature called "tools". When building with `--no-default-features` flag most of the tools included in the regular binary won't be included (TS support, doc generator, formatter, linter, bundler, file watcher) and produced binary will be able to load file only from local disk (so effectively no URL support and only "deno run" subcommand is operational). This reduces binary size from about 45Mb to 30Mb (on macOS).

There are still about 50 "unused" warnings left.

To try out locally:
```
cd cli  // this is required - cannot build workspace with features
cargo build --no-default-features
```

### CODE REVIEW

**Comment by bartlomieju** [comment]
Update:

When building with `--no-default-features` results as follows:
- release binary 27Mb; zipped 14Mb
- release binary without debug info 16Mb; zipped 6.8Mb

File: cli/ast.rs [hunk 1]
```diff
@@ -135,8 +135,8 @@ impl DiagnosticBuffer {
 #[derive(Debug, Clone)]
 pub struct ErrorBuffer(Arc<RwLock<Vec<Diagnostic>>>);
 
-impl ErrorBuffer {
-  pub fn new() -> Self {
```
Comment (ry) [inline cli/ast.rs:139]: aside: shouldn't this result in a clippy error currently?

Comment (bartlomieju) [reply cli/ast.rs:139]: I would expect so, but it seems that's only the case when having `lib.rs`

**Comment by bartlomieju** [comment]
Closing in favor of #8640

**Pull request closed without merging**